src/praisonai-agents/praisonaiagents/agent/tool_execution.py

```
@@ -300,6 +300,47 @@ def execute_with_context():
+                        else:
+                            # For successful tools, use the normalized content
+                            result = normalized_result.content
+                        
+                        # Store normalized result for hooks to access full context
+                        self._last_normalized_result = normalized_result
+                        
+                        logger.debug(f"Applied runtime middleware for {runtime_id}: {function_name}")
+                except ImportError:
+                    # Runtime middleware not available - continue without normalization
+                    logger.debug("Runtime middleware not available, skipping normalization")
+                except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)